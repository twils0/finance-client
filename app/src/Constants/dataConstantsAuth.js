export const actionTypes = {
  SET_AUTHENTICATED: 'DATA_AUTH_SET_AUTHENTICATED',
  SET_REDIRECT_URL: 'DATA_AUTH_SET_REDIRECT_URL',
  SET_AUTH_STATUS: 'DATA_AUTH_SET_AUTH_STATUS',
  SET_CODE_TYPE: 'DATA_AUTH_SET_CODE_TYPE',
  RESET_AUTH_STATE: 'DATA_AUTH_RESET_AUTH_STATE',
};

export const statusNames = {
  LOGIN: 'authStatusLogin',
  SIGN_OUT_DEVICES: 'authStatusSignOutDevices',
  LOGIN_MFA: 'authStatusLoginMFA',
  SIGN_UP: 'authStatusSignUp',
  VERIFY_PHONE: 'authStatusVerifyPhone',
  VERIFY_PHONE_CODE: 'authStatusVerifyPhoneCode',
  VERIFY_EMAIL: 'authStatusVerifyEmail',
  VERIFY_EMAIL_LINK: 'authStatusVerifyEmailLink',
  FORGOT_PASSWORD: 'authStatusForgotPassword',
  RESET_PASSWORD: 'authStatusResetPassword',
  CHANGE_PASSWORD: 'authStatusChangePassword',
  DELETE_ACCOUNT: 'authStatusDeleteAccount',
  LOGOUT: 'authStatusLogout',
};

export const codeTypeNames = {
  VERIFY_PHONE: 'verifyPhone',
  VERIFY_EMAIL: 'verifyEmail',
  VERIFY_EMAIL_ADDITIONAL: 'verifyEmailAdditional',
};
