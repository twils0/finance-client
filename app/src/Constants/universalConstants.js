import stripeConfig from '../../../stripe.config.json';
import dbConfig from '../../../db.config.json';
import iexConfig from '../../../iex.config.json';

export const baseURLs = {
  STRIPE: stripeConfig.url,
  DB: dbConfig.url,
  IEX: iexConfig.urls.api,
  IEX_DEV: iexConfig.urls.dev,
  IEX_TERMS: iexConfig.urls.terms,
};

export const URLs = {
  CUSTOMERS: '/customers',
  USERS: '/users',
  EMAILS: '/emails',
  SECURITIES: '/securities',
  SECURITIES_SEARCH: '/securities/search',
  IEX_COMPANY: tickerCusip => `/stock/${tickerCusip}/company`,
  IEX_QUOTE: tickerCusip => `/stock/${tickerCusip}/quote`,
};

export const axiosConfig = {
  STRIPE: {
    baseURL: baseURLs.STRIPE,
    timeout: 30000,
  },
  DB: {
    baseURL: baseURLs.DB,
    timeout: 30000,
  },
  IEX: {
    baseURL: baseURLs.IEX,
    timeout: 30000,
  },
};

export const pathNames = {
  DEFAULT: '*',
  LOGIN: '/login',
  SIGN_UP: '/signup',
  CODE: '/code',
  CODE_VERIFY_EMAIL: '/code/:verificationId?',
  TERMS: '/terms',
  INTERNAL: '/internal',
  WATCHLIST: '/internal/watchlist',
  WATCHLIST_SECURITY_ID: '/internal/watchlist/:securityId?',
  ACCOUNT: '/internal/account',
};

export const requestStatusTypes = {
  IDLE: 'IDLE',
  LOADING: 'LOADING',
  SUCCESS: 'SUCCESS',
  ERROR: 'ERROR',
};
