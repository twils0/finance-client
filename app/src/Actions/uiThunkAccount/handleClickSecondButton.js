import { codeTypeNames } from '../../Constants/dataConstantsAuth';
import { formNames } from '../../Constants/uiConstantsAccount';
import requestVerifyField from '../dataThunkAuth/requestVerifyField';
import handleCancel from '../uiThunkAccount/handleCancel';

const handleClickSecondButton = (payload) => {
  if (!Object.prototype.hasOwnProperty.call(payload, 'resetStripeElement')) {
    throw new Error(
      `Please enter a value for the 'resetStripeElement' key - ${JSON.stringify(payload)}`,
    );
  }

  return (dispatch, getState) => {
    const state = getState();
    const { forms } = state.ui.internal.account;
    const { codeTypes } = state.data.auth;

    switch (forms.current) {
      case formNames.CODE: {
        if (codeTypes[codeTypeNames.VERIFY_PHONE].needed) {
          dispatch(requestVerifyField({ field: 'phone_number' }));
        }
        break;
      }
      case formNames.BILLING: {
        payload.resetStripeElement();
        break;
      }
      default:
        break;
    }

    dispatch(handleCancel());

    return null;
  };
};

export default handleClickSecondButton;
