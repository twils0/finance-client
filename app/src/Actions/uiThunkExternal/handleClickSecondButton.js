import { formNames } from '../../Constants/uiConstantsExternal';
import { setCurrentForm } from '../uiActionsExternal';

import secondButtonResetPassword from './secondButtonResetPassword';
import secondButtonForgotPassword from './secondButtonForgotPassword';
import secondButtonLogin from './secondButtonLogin';
import secondButtonCard from './secondButtonCard';
import secondButtonSignUp from './secondButtonSignUp';
import secondButtonCodeMFAPhone from './secondButtonCodeMFAPhone';
import secondButtonCodeVerifyEmail from './secondButtonCodeVerifyEmail';
import secondButtonDevice from './secondButtonDevice';

const handleClickSecondButton = (payload) => {
  if (!Object.prototype.hasOwnProperty.call(payload, 'history')) {
    throw new Error(`Please enter a value for the 'history' key - ${JSON.stringify(payload)}`);
  }
  if (!Object.prototype.hasOwnProperty.call(payload, 'resetStripeElement')) {
    throw new Error(
      `Please enter a value for the 'resetStripeElement' key - ${JSON.stringify(payload)}`,
    );
  }

  return (dispatch, getState) => {
    const state = getState();
    const { forms } = state.ui.external;

    switch (forms.current) {
      case formNames.RESET_PASSWORD:
        dispatch(secondButtonResetPassword());
        break;
      case formNames.FORGOT_PASSWORD:
        dispatch(secondButtonForgotPassword());
        break;
      case formNames.LOGIN: {
        const { history } = payload;

        dispatch(secondButtonLogin({ history }));
        break;
      }
      case formNames.CARD: {
        const { history, resetStripeElement } = payload;

        dispatch(secondButtonCard({ history, resetStripeElement }));
        break;
      }
      case formNames.INFO:
        dispatch(setCurrentForm({ current: formNames.CARD }));
        break;
      case formNames.SIGN_UP:
        dispatch(secondButtonSignUp());
        break;
      case formNames.CODE_MFA_PHONE: {
        const { history } = payload;

        dispatch(secondButtonCodeMFAPhone({ history }));
        break;
      }
      case formNames.CODE_VERIFY_EMAIL: {
        const { history } = payload;

        dispatch(secondButtonCodeVerifyEmail({ history }));
        break;
      }
      case formNames.DEVICE: {
        const { history } = payload;

        dispatch(secondButtonDevice({ history }));
        break;
      }
      default:
        break;
    }

    return null;
  };
};

export default handleClickSecondButton;
