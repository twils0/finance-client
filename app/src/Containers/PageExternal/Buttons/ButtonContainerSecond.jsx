import React from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';

import { requestStatusTypes } from '../../../Constants/universalConstants';
import { statusNames } from '../../../Constants/dataConstantsAuth';
import { buttonNames } from '../../../Constants/uiConstantsExternal';
import handleClickSecondButton from '../../../Actions/uiThunkExternal/handleClickSecondButton';

import ButtonSecond from '../../../Components/PageExternal/Buttons/ButtonSecond';

class ButtonContainerSecond extends React.Component {
  handleClick = (clickEvent) => {
    clickEvent.preventDefault();
    const { handleClickButton, history, resetStripeElement } = this.props;

    handleClickButton({ history, resetStripeElement });
  };

  render() {
    const { width, statusAuth, text } = this.props;
    let disabled = false;

    if (
      statusAuth[statusNames.LOGIN].status === requestStatusTypes.LOADING
      || statusAuth[statusNames.SIGN_OUT_DEVICES].status === requestStatusTypes.LOADING
      || statusAuth[statusNames.LOGIN_MFA].status === requestStatusTypes.LOADING
      || statusAuth[statusNames.VERIFY_EMAIL].status === requestStatusTypes.LOADING
      || statusAuth[statusNames.FORGOT_PASSWORD].status === requestStatusTypes.LOADING
      || statusAuth[statusNames.CHANGE_PASSWORD].status === requestStatusTypes.LOADING
      || statusAuth[statusNames.SIGN_UP].status === requestStatusTypes.LOADING
      || statusAuth[statusNames.SIGN_UP].status === requestStatusTypes.SUCCESS
    ) {
      disabled = true;
    }

    return (
      <ButtonSecond width={width} disabled={disabled} text={text} handleClick={this.handleClick} />
    );
  }
}

ButtonContainerSecond.propTypes = {
  history: PropTypes.object.isRequired,
  width: PropTypes.number.isRequired,
  text: PropTypes.string.isRequired,
  statusAuth: PropTypes.object.isRequired,
  resetStripeElement: PropTypes.func.isRequired,
  handleClickButton: PropTypes.func.isRequired,
};

ButtonContainerSecond.defaultProps = {};

const mapStateToProps = (state, ownProps) => ({
  width: state.ui.external.utilities.buttons.width,
  heightBuffer: state.ui.external.utilities.buttons.heightBuffer,
  forms: state.ui.external.forms,
  text: state.ui.external.buttons[buttonNames.SECOND].text,
  statusAuth: state.data.auth.status,
  resetStripeElement: ownProps.resetStripeElement,
});

const mapDispatchToProps = dispatch => ({
  handleClickButton: payload => dispatch(handleClickSecondButton(payload)),
});

export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps,
  )(ButtonContainerSecond),
);
