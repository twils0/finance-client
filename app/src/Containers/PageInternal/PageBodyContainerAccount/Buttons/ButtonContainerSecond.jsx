import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { requestStatusTypes } from '../../../../Constants/universalConstants';
import { statusNames } from '../../../../Constants/dataConstantsAccount';
import { buttonNames } from '../../../../Constants/uiConstantsAccount';

import handleClickSecondButton from '../../../../Actions/uiThunkAccount/handleClickSecondButton';

import ButtonSecond from '../../../../Components/PageInternal/PageBodyAccount/Buttons/ButtonSecond';

class ButtonContainerSecond extends React.Component {
  handleClick = (clickEvent) => {
    clickEvent.preventDefault();
    const { handleClickButton, resetStripeElement } = this.props;

    handleClickButton({ resetStripeElement });
  };

  render() {
    const {
      statusAccount, width, heightBuffer, secondButtonText,
    } = this.props;
    let disabled = false;

    if (
      statusAccount[statusNames.UPDATE_STRIPE_FIELDS_REQUEST].status ===
        requestStatusTypes.LOADING ||
      statusAccount[statusNames.UPDATE_DB_FIELDS].status === requestStatusTypes.LOADING ||
      statusAccount[statusNames.UPDATE_AWS_FIELDS].status === requestStatusTypes.LOADING
    ) {
      disabled = true;
    }

    return (
      <ButtonSecond
        width={width}
        heightBuffer={heightBuffer}
        disabled={disabled}
        text={secondButtonText}
        handleClick={this.handleClick}
      />
    );
  }
}

ButtonContainerSecond.propTypes = {
  width: PropTypes.number.isRequired,
  heightBuffer: PropTypes.number.isRequired,
  statusAccount: PropTypes.object.isRequired,
  secondButtonText: PropTypes.string.isRequired,
  resetStripeElement: PropTypes.func.isRequired,
  handleClickButton: PropTypes.func.isRequired,
};

ButtonContainerSecond.defaultProps = {};

const mapStateToProps = (state, ownProps) => ({
  width: state.ui.internal.account.utilities.buttons.width,
  heightBuffer: state.ui.internal.account.utilities.buttons.heightBuffer,
  statusAccount: state.data.account.status,
  secondButtonText: state.ui.internal.account.buttons[buttonNames.SECOND].text,
  resetStripeElement: ownProps.resetStripeElement,
});

const mapDispatchToProps = dispatch => ({
  handleClickButton: payload => dispatch(handleClickSecondButton(payload)),
});

export default connect(mapStateToProps, mapDispatchToProps)(ButtonContainerSecond);
