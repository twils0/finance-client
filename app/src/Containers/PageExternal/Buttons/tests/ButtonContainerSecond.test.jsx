import React from 'react';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { shallow } from 'enzyme';

import { initialState as initialStateAuth } from '../../../../Reducers/dataReducersAuth';
import { initialState as initialStateUIExternal } from '../../../../Reducers/uiReducersExternal';
import { requestStatusTypes } from '../../../../Constants/universalConstants';
import { statusNames } from '../../../../Constants/dataConstantsAuth';

import ButtonContainerSecond from '../ButtonContainerSecond';

import handleClickSecondButton from '../../../../Actions/uiThunkExternal/handleClickSecondButton';

const middleware = [thunk];
const mockStore = configureMockStore(middleware);

jest.mock('../../../../Actions/uiThunkExternal/handleClickSecondButton', () => jest.fn());
handleClickSecondButton.mockReturnValue(() => null);

const history = { test: 'testHistory' };

const clickEvent = {
  preventDefault: jest.fn(),
};
const resetStripeElement = jest.fn();

const shallowComponent = store =>
  shallow(<ButtonContainerSecond resetStripeElement={resetStripeElement} />, {
    context: {
      store,
      router: {
        history,
        route: {
          location: {},
          match: {},
        },
      },
    },
  })
    .dive()
    .dive()
    .dive();

describe('Containers', () => {
  describe('PageExternal', () => {
    describe('Buttons', () => {
      describe('ButtonContainerSecond', () => {
        afterEach(() => {
          handleClickSecondButton.mockReset();
        });

        it('shallow renders correctly and button works, with disabled false', async () => {
          const store = mockStore({
            data: {
              auth: initialStateAuth,
            },
            ui: {
              external: initialStateUIExternal,
            },
          });

          const wrapper = shallowComponent(store);

          const instance = wrapper.instance();
          const props = wrapper.props();

          instance.handleClick(clickEvent);

          expect(wrapper).toMatchSnapshot();
          expect(props.disabled).toEqual(false);
          expect(handleClickSecondButton).toBeCalledWith({
            history,
            resetStripeElement,
          });
        });

        it('shallow renders correctly, with disabled true', async () => {
          const stateBeforeAuth = JSON.parse(JSON.stringify(initialStateAuth));
          stateBeforeAuth.status[statusNames.SIGN_UP].status = requestStatusTypes.LOADING;

          const store = mockStore({
            data: {
              auth: stateBeforeAuth,
            },
            ui: {
              external: initialStateUIExternal,
            },
          });

          const wrapper = shallowComponent(store);

          const props = wrapper.props();

          expect(wrapper).toMatchSnapshot();
          expect(props.disabled).toEqual(true);
        });
      });
    });
  });
});
