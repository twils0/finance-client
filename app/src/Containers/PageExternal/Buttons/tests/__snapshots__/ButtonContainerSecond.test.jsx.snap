// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Containers PageExternal Buttons ButtonContainerSecond shallow renders correctly and button works, with disabled false 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ButtonContainerSecond
    forms={
      Object {
        "cardForm": Object {
          "height": 360,
          "id": "cardForm",
          "inputs": Object {
            "cardNameOnCard": Object {
              "errorMessage": "",
              "id": "cardNameOnCard",
              "parentId": "cardForm",
              "value": "",
            },
            "cardPromoCode": Object {
              "errorMessage": "",
              "id": "cardPromoCode",
              "parentId": "cardForm",
              "value": "",
            },
            "cardStripe": Object {
              "errorMessage": "",
              "id": "cardStripe",
              "parentId": "cardForm",
              "value": "",
            },
          },
          "inputsList": Array [
            "cardNameOnCard",
            "cardPromoCode",
            "cardStripe",
          ],
        },
        "codeEmailVerifyForm": Object {
          "height": 183,
          "id": "codeEmailVerifyForm",
          "inputs": Object {
            "codeVerifyEmailCode": Object {
              "errorMessage": "",
              "id": "codeVerifyEmailCode",
              "parentId": "codeEmailVerifyForm",
              "value": "",
            },
          },
          "inputsList": Array [
            "codeVerifyEmailCode",
          ],
        },
        "codePhoneMFAForm": Object {
          "height": 183,
          "id": "codePhoneMFAForm",
          "inputs": Object {
            "codeMFAPhoneCode": Object {
              "errorMessage": "",
              "id": "codeMFAPhoneCode",
              "parentId": "codePhoneMFAForm",
              "value": "",
            },
          },
          "inputsList": Array [
            "codeMFAPhoneCode",
          ],
        },
        "codePhoneVerifyForm": Object {
          "height": 183,
          "id": "codePhoneVerifyForm",
          "inputs": Object {
            "codeVerifyPhoneCode": Object {
              "errorMessage": "",
              "id": "codeVerifyPhoneCode",
              "parentId": "codePhoneVerifyForm",
              "value": "",
            },
          },
          "inputsList": Array [
            "codeVerifyPhoneCode",
          ],
        },
        "current": "loginForm",
        "deviceForm": Object {
          "height": 146,
          "id": "deviceForm",
          "inputs": Object {},
          "inputsList": Array [],
        },
        "forgotPasswordForm": Object {
          "height": 220,
          "id": "forgotPasswordForm",
          "inputs": Object {
            "loginPasswordEmail": Object {
              "errorMessage": "",
              "id": "loginPasswordEmail",
              "parentId": "forgotPasswordForm",
              "value": "",
            },
          },
          "inputsList": Array [],
        },
        "infoForm": Object {
          "height": 280,
          "id": "infoForm",
          "inputs": Object {
            "infoEmail": Object {
              "errorMessage": "",
              "id": "infoEmail",
              "parentId": "infoForm",
              "value": "",
            },
            "infoName": Object {
              "errorMessage": "",
              "id": "infoName",
              "parentId": "infoForm",
              "value": "",
            },
            "infoPhone": Object {
              "errorMessage": "",
              "id": "infoPhone",
              "parentId": "infoForm",
              "value": "",
            },
          },
          "inputsList": Array [
            "infoName",
            "infoPhone",
            "infoEmail",
          ],
        },
        "list": Array [
          "resetPasswordForm",
          "forgotPasswordForm",
          "loginForm",
          "cardForm",
          "infoForm",
          "signUpForm",
          "codePhoneMFAForm",
          "codePhoneVerifyForm",
          "codeEmailVerifyForm",
          "deviceForm",
        ],
        "loginForm": Object {
          "height": 222,
          "id": "loginForm",
          "inputs": Object {
            "loginEmail": Object {
              "errorMessage": "",
              "id": "loginEmail",
              "parentId": "loginForm",
              "value": "",
            },
            "loginPassword": Object {
              "errorMessage": "",
              "id": "loginPassword",
              "parentId": "loginForm",
              "value": "",
            },
          },
          "inputsList": Array [
            "loginEmail",
            "loginPassword",
          ],
        },
        "resetPasswordForm": Object {
          "height": 339,
          "id": "resetPasswordForm",
          "inputs": Object {
            "codeResetPassword": Object {
              "errorMessage": "",
              "id": "codeResetPassword",
              "parentId": "resetPasswordForm",
              "value": "",
            },
            "password2ResetPassword": Object {
              "errorMessage": "",
              "id": "password2ResetPassword",
              "parentId": "resetPasswordForm",
              "value": "",
            },
            "passwordResetPassword": Object {
              "errorMessage": "",
              "id": "passwordResetPassword",
              "parentId": "resetPasswordForm",
              "value": "",
            },
          },
          "inputsList": Array [
            "codeResetPassword",
            "passwordResetPassword",
            "password2ResetPassword",
          ],
        },
        "signUpForm": Object {
          "height": 247,
          "id": "signUpForm",
          "inputs": Object {
            "signUpPassword": Object {
              "errorMessage": "",
              "id": "signUpPassword",
              "parentId": "signUpForm",
              "value": "",
            },
            "signUpPassword2": Object {
              "errorMessage": "",
              "id": "signUpPassword2",
              "parentId": "signUpForm",
              "value": "",
            },
          },
          "inputsList": Array [
            "signUpPassword",
            "signUpPassword2",
          ],
        },
      }
    }
    handleClickButton={[Function]}
    heightBuffer={20}
    history={
      Object {
        "test": "testHistory",
      }
    }
    location={Object {}}
    match={Object {}}
    resetStripeElement={[MockFunction]}
    statusAuth={
      Object {
        "authStatusChangePassword": Object {
          "id": "authStatusChangePassword",
          "status": "IDLE",
        },
        "authStatusDeleteAccount": Object {
          "id": "authStatusDeleteAccount",
          "status": "SUCCESS",
        },
        "authStatusForgotPassword": Object {
          "id": "authStatusForgotPassword",
          "status": "IDLE",
        },
        "authStatusLogin": Object {
          "id": "authStatusLogin",
          "status": "IDLE",
        },
        "authStatusLoginMFA": Object {
          "id": "authStatusLoginMFA",
          "status": "IDLE",
        },
        "authStatusLogout": Object {
          "id": "authStatusLogout",
          "status": "SUCCESS",
        },
        "authStatusResetPassword": Object {
          "id": "authStatusResetPassword",
          "status": "IDLE",
        },
        "authStatusSignOutDevices": Object {
          "id": "authStatusSignOutDevices",
          "status": "IDLE",
        },
        "authStatusSignUp": Object {
          "id": "authStatusSignUp",
          "status": "IDLE",
        },
        "authStatusVerifyEmail": Object {
          "id": "authStatusVerifyEmail",
          "status": "IDLE",
        },
        "authStatusVerifyEmailLink": Object {
          "id": "authStatusVerifyEmailLink",
          "status": "IDLE",
        },
        "authStatusVerifyPhone": Object {
          "id": "authStatusVerifyPhone",
          "status": "IDLE",
        },
        "authStatusVerifyPhoneCode": Object {
          "id": "authStatusVerifyPhoneCode",
          "status": "IDLE",
        },
      }
    }
    text="Sign Up"
    width={300}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "disabled": false,
      "handleClick": [Function],
      "text": "Sign Up",
      "width": 300,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "disabled": false,
        "handleClick": [Function],
        "text": "Sign Up",
        "width": 300,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "context": Object {
      "router": Object {
        "history": Object {
          "test": "testHistory",
        },
        "route": Object {
          "location": Object {},
          "match": Object {},
        },
      },
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
    "hydrateIn": undefined,
  },
}
`;

exports[`Containers PageExternal Buttons ButtonContainerSecond shallow renders correctly, with disabled true 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ButtonContainerSecond
    forms={
      Object {
        "cardForm": Object {
          "height": 360,
          "id": "cardForm",
          "inputs": Object {
            "cardNameOnCard": Object {
              "errorMessage": "",
              "id": "cardNameOnCard",
              "parentId": "cardForm",
              "value": "",
            },
            "cardPromoCode": Object {
              "errorMessage": "",
              "id": "cardPromoCode",
              "parentId": "cardForm",
              "value": "",
            },
            "cardStripe": Object {
              "errorMessage": "",
              "id": "cardStripe",
              "parentId": "cardForm",
              "value": "",
            },
          },
          "inputsList": Array [
            "cardNameOnCard",
            "cardPromoCode",
            "cardStripe",
          ],
        },
        "codeEmailVerifyForm": Object {
          "height": 183,
          "id": "codeEmailVerifyForm",
          "inputs": Object {
            "codeVerifyEmailCode": Object {
              "errorMessage": "",
              "id": "codeVerifyEmailCode",
              "parentId": "codeEmailVerifyForm",
              "value": "",
            },
          },
          "inputsList": Array [
            "codeVerifyEmailCode",
          ],
        },
        "codePhoneMFAForm": Object {
          "height": 183,
          "id": "codePhoneMFAForm",
          "inputs": Object {
            "codeMFAPhoneCode": Object {
              "errorMessage": "",
              "id": "codeMFAPhoneCode",
              "parentId": "codePhoneMFAForm",
              "value": "",
            },
          },
          "inputsList": Array [
            "codeMFAPhoneCode",
          ],
        },
        "codePhoneVerifyForm": Object {
          "height": 183,
          "id": "codePhoneVerifyForm",
          "inputs": Object {
            "codeVerifyPhoneCode": Object {
              "errorMessage": "",
              "id": "codeVerifyPhoneCode",
              "parentId": "codePhoneVerifyForm",
              "value": "",
            },
          },
          "inputsList": Array [
            "codeVerifyPhoneCode",
          ],
        },
        "current": "loginForm",
        "deviceForm": Object {
          "height": 146,
          "id": "deviceForm",
          "inputs": Object {},
          "inputsList": Array [],
        },
        "forgotPasswordForm": Object {
          "height": 220,
          "id": "forgotPasswordForm",
          "inputs": Object {
            "loginPasswordEmail": Object {
              "errorMessage": "",
              "id": "loginPasswordEmail",
              "parentId": "forgotPasswordForm",
              "value": "",
            },
          },
          "inputsList": Array [],
        },
        "infoForm": Object {
          "height": 280,
          "id": "infoForm",
          "inputs": Object {
            "infoEmail": Object {
              "errorMessage": "",
              "id": "infoEmail",
              "parentId": "infoForm",
              "value": "",
            },
            "infoName": Object {
              "errorMessage": "",
              "id": "infoName",
              "parentId": "infoForm",
              "value": "",
            },
            "infoPhone": Object {
              "errorMessage": "",
              "id": "infoPhone",
              "parentId": "infoForm",
              "value": "",
            },
          },
          "inputsList": Array [
            "infoName",
            "infoPhone",
            "infoEmail",
          ],
        },
        "list": Array [
          "resetPasswordForm",
          "forgotPasswordForm",
          "loginForm",
          "cardForm",
          "infoForm",
          "signUpForm",
          "codePhoneMFAForm",
          "codePhoneVerifyForm",
          "codeEmailVerifyForm",
          "deviceForm",
        ],
        "loginForm": Object {
          "height": 222,
          "id": "loginForm",
          "inputs": Object {
            "loginEmail": Object {
              "errorMessage": "",
              "id": "loginEmail",
              "parentId": "loginForm",
              "value": "",
            },
            "loginPassword": Object {
              "errorMessage": "",
              "id": "loginPassword",
              "parentId": "loginForm",
              "value": "",
            },
          },
          "inputsList": Array [
            "loginEmail",
            "loginPassword",
          ],
        },
        "resetPasswordForm": Object {
          "height": 339,
          "id": "resetPasswordForm",
          "inputs": Object {
            "codeResetPassword": Object {
              "errorMessage": "",
              "id": "codeResetPassword",
              "parentId": "resetPasswordForm",
              "value": "",
            },
            "password2ResetPassword": Object {
              "errorMessage": "",
              "id": "password2ResetPassword",
              "parentId": "resetPasswordForm",
              "value": "",
            },
            "passwordResetPassword": Object {
              "errorMessage": "",
              "id": "passwordResetPassword",
              "parentId": "resetPasswordForm",
              "value": "",
            },
          },
          "inputsList": Array [
            "codeResetPassword",
            "passwordResetPassword",
            "password2ResetPassword",
          ],
        },
        "signUpForm": Object {
          "height": 247,
          "id": "signUpForm",
          "inputs": Object {
            "signUpPassword": Object {
              "errorMessage": "",
              "id": "signUpPassword",
              "parentId": "signUpForm",
              "value": "",
            },
            "signUpPassword2": Object {
              "errorMessage": "",
              "id": "signUpPassword2",
              "parentId": "signUpForm",
              "value": "",
            },
          },
          "inputsList": Array [
            "signUpPassword",
            "signUpPassword2",
          ],
        },
      }
    }
    handleClickButton={[Function]}
    heightBuffer={20}
    history={
      Object {
        "test": "testHistory",
      }
    }
    location={Object {}}
    match={Object {}}
    resetStripeElement={[MockFunction]}
    statusAuth={
      Object {
        "authStatusChangePassword": Object {
          "id": "authStatusChangePassword",
          "status": "IDLE",
        },
        "authStatusDeleteAccount": Object {
          "id": "authStatusDeleteAccount",
          "status": "SUCCESS",
        },
        "authStatusForgotPassword": Object {
          "id": "authStatusForgotPassword",
          "status": "IDLE",
        },
        "authStatusLogin": Object {
          "id": "authStatusLogin",
          "status": "IDLE",
        },
        "authStatusLoginMFA": Object {
          "id": "authStatusLoginMFA",
          "status": "IDLE",
        },
        "authStatusLogout": Object {
          "id": "authStatusLogout",
          "status": "SUCCESS",
        },
        "authStatusResetPassword": Object {
          "id": "authStatusResetPassword",
          "status": "IDLE",
        },
        "authStatusSignOutDevices": Object {
          "id": "authStatusSignOutDevices",
          "status": "IDLE",
        },
        "authStatusSignUp": Object {
          "id": "authStatusSignUp",
          "status": "LOADING",
        },
        "authStatusVerifyEmail": Object {
          "id": "authStatusVerifyEmail",
          "status": "IDLE",
        },
        "authStatusVerifyEmailLink": Object {
          "id": "authStatusVerifyEmailLink",
          "status": "IDLE",
        },
        "authStatusVerifyPhone": Object {
          "id": "authStatusVerifyPhone",
          "status": "IDLE",
        },
        "authStatusVerifyPhoneCode": Object {
          "id": "authStatusVerifyPhoneCode",
          "status": "IDLE",
        },
      }
    }
    text="Sign Up"
    width={300}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "disabled": true,
      "handleClick": [Function],
      "text": "Sign Up",
      "width": 300,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "disabled": true,
        "handleClick": [Function],
        "text": "Sign Up",
        "width": 300,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "context": Object {
      "router": Object {
        "history": Object {
          "test": "testHistory",
        },
        "route": Object {
          "location": Object {},
          "match": Object {},
        },
      },
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
    "hydrateIn": undefined,
  },
}
`;
